diff --git a/themes/af-magic.zsh-theme b/themes/af-magic.zsh-theme
index 1b629e4..f9b5836 100644
--- a/themes/af-magic.zsh-theme
+++ b/themes/af-magic.zsh-theme
@@ -3,37 +3,65 @@
 # Author: Andy Fleming
 # URL: http://andyfleming.com/
 
+timer_preexec() {
+  timer=$(date +%s%0N)
+}
+
+timer_precmd() {
+  if [ $timer ]; then
+    local now=$(date +%s%0N)
+    local elapsed=$((now-timer))
+
+    local mins=$(printf '%d' $((elapsed / 60 / 1000000000)))
+    local secs=$(bc <<< "scale=3;$elapsed / 1000000000 - 60 * $mins")
+    secs=$(printf "%0.3f" secs)
+    secs=${secs/,/.}
+
+    if [ $mins -gt 0 ]; then
+      timestring="{${mins}m ${secs}s}"
+    else
+      timestring="{${secs}s}"
+    fi
+
+    unset timer
+  else
+    unset timestring
+  fi
+}
+
+add-zsh-hook preexec timer_preexec
+add-zsh-hook precmd timer_precmd
+
+
 # dashed separator size
 function afmagic_dashes {
-  # check either virtualenv or condaenv variables
-  local python_env="${VIRTUAL_ENV:-$CONDA_DEFAULT_ENV}"
+  local pref="$(print -P %~)"
 
-  # if there is a python virtual environment and it is displayed in
-  # the prompt, account for it when returning the number of dashes
-  if [[ -n "$python_env" && "$PS1" = *\(${python_env}\)* ]]; then
-    echo $(( COLUMNS - ${#python_env} - 3 ))
+  if [ $timestring ]; then
+    local suff="$timestring"
+    echo $(( COLUMNS - ${#pref} - ${#suff} - 3 ))
   else
-    echo $COLUMNS
+    echo $(( COLUMNS - ${#pref} - 2 ))
   fi
 }
 
+
 # primary prompt: dashed separator, directory and vcs info
-PS1="${FG[237]}\${(l.\$(afmagic_dashes)..-.)}%{$reset_color%}
-${FG[032]}%~\$(git_prompt_info)\$(hg_prompt_info) ${FG[105]}%(!.#.»)%{$reset_color%} "
+PS1="${FG[032]}\$(print -P %~) ${FG[237]}\${(l.\$(afmagic_dashes)..-.)} ${FG[172]}\$timestring%{$reset_color%}
+${FG[105]}%(!.#.»)%{$reset_color%} "
 PS2="%{$fg[red]%}\ %{$reset_color%}"
 
 # right prompt: return code, virtualenv and context (user@host)
-RPS1="%(?..%{$fg[red]%}%? ↵%{$reset_color%})"
+RPS1="%(?..%{$fg[red]%}%? ↵%{$reset_color%}) "
 if (( $+functions[virtualenv_prompt_info] )); then
   RPS1+='$(virtualenv_prompt_info)'
 fi
-RPS1+=" ${FG[237]}%n@%m%{$reset_color%}"
 
 # git settings
-ZSH_THEME_GIT_PROMPT_PREFIX=" ${FG[075]}(${FG[078]}"
-ZSH_THEME_GIT_PROMPT_CLEAN=""
-ZSH_THEME_GIT_PROMPT_DIRTY="${FG[214]}*%{$reset_color%}"
-ZSH_THEME_GIT_PROMPT_SUFFIX="${FG[075]})%{$reset_color%}"
+# ZSH_THEME_GIT_PROMPT_PREFIX=" ${FG[075]}(${FG[078]}"
+# ZSH_THEME_GIT_PROMPT_CLEAN=""
+# ZSH_THEME_GIT_PROMPT_DIRTY="${FG[214]}*%{$reset_color%}"
+# ZSH_THEME_GIT_PROMPT_SUFFIX="${FG[075]})%{$reset_color%}"
 
 # hg settings
 ZSH_THEME_HG_PROMPT_PREFIX=" ${FG[075]}(${FG[078]}"
